src/helpers/staking.js

```diff
@@ -481,15 +481,19 @@ export const fetchStakingData = async (eth) => {
 
   _stakingData.rewards = rewards.sort((rewardA, rewardB) => rewardB.timestamp - rewardA.timestamp);
 
-  // retrieving the votes in the last month for a given address...
-  _stakingData.votes = await fetchLastMonthVoteForVoter(eth.address);
-  
-  // retrieving the oldest active proposal from piedao.eth space after the 18/10/2021...
-  _stakingData.proposals = await fetchLastSnapshots(1, 'active', 'asc', moment("2021-10-18").unix());
-  // and if there is at least one active proposal after the 18/10/2021, we add the
-  // block infos into that object, so we can easily get the timestamp or any other related info
-  if(_stakingData.proposals[0]) {
-    _stakingData.proposals[0].block = await eth.provider.getBlock(Number(_stakingData.proposals[0].snapshot));
+  try {
+    // retrieving the votes in the last month for a given address...
+    _stakingData.votes = await fetchLastMonthVoteForVoter(eth.address);
+    
+    // retrieving the oldest active proposal from piedao.eth space after the 18/10/2021...
+    _stakingData.proposals = await fetchLastSnapshots(1, 'active', 'asc', moment("2021-10-18").unix());
+    // and if there is at least one active proposal after the 18/10/2021, we add the
+    // block infos into that object, so we can easily get the timestamp or any other related info
+    if(_stakingData.proposals[0]) {
+      _stakingData.proposals[0].block = await eth.provider.getBlock(Number(_stakingData.proposals[0].snapshot));
+    }    
+  } catch(error) {
+    console.error('staking - snapshot error', error);
```

Comment:
This needs to be visual, use display notification

Reply:
@Alexintosh this will run every 5 seconds, if i put it visual we will have a terrible UX

Reply:
@Alexintosh as discussed on discord, let's revise this in a very near future.